e2e-tests-rust/src/provider/testing.rs

```diff
@@ -93,6 +101,66 @@ pub async fn init_testing_provider(
     })
 }
 
+// Init testing provider which sends specified HTTP headers e.g. for authentication
+// Outside of `TestingProvider` to avoid specifying `P`
+pub async fn init_testing_provider_with_http_headers(
```

Comment:
Also, isn't it possible to express the logic of adding headers as a `fn add_headers(EraTestNode) -> EraTestNode`, so that this method is not needed at all? Then you will be able to use `init_testing_provider(add_headers)` or smth.

Note that function is composable, so we can wrap any number of layers here, e.g. `init_testing_provider(|node| f1(f2(node)))`

Reply:
No, because it's not the setting on the node, it's the setting on provider.

Reply:
We could extend the provider code of `on_era_test_node_with_wallet_and_config` or add another one on the `alloy-zksync` side, but didn't want to do that as not sure if it is useful for any other cases.